tests/utils/common/common.py

```diff
@@ -0,0 +1,615 @@
+        finally:
+            # Restore the original connection parameters
+            conn.connect_timeout = orig_timeout
+
+
+def determine_active_passive_part(bitbake_variables, conn):
+    """Given the output from mount, determine the currently active and passive
+    partitions, returning them as a pair in that order."""
+
+    mount_output = conn.run("mount").stdout
+    a = bitbake_variables["MENDER_ROOTFS_PART_A"]
+    b = bitbake_variables["MENDER_ROOTFS_PART_B"]
+
+    if mount_output.find(a) >= 0:
+        return (a, b)
+    elif mount_output.find(b) >= 0:
+        return (b, a)
+    else:
+        raise Exception(
+            "Could not determine active partition. Mount output:\n {}"
+            "\nwas looking for {}".format(mount_output, (a, b))
+        )
+
+
+def get_ssh_common_args(conn):
+    args = "-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+    if "key_filename" in conn.connect_kwargs.keys():
+        args += " -i %s" % conn.connect_kwargs["key_filename"]
+    return args
+
+
+# Yocto build SSH is lacking SFTP, let's override and use regular SCP instead.
+def put_no_sftp(file, conn, remote="."):
+    cmd = "scp %s" % get_ssh_common_args(conn)
+    conn.local(
+        "%s -P %s %s %s@%s:%s" % (cmd, conn.port, file, conn.user, conn.host, remote)
+    )
+
+
+# Yocto build SSH is lacking SFTP, let's override and use regular SCP instead.
+def get_no_sftp(file, conn, local="."):
+    cmd = "scp %s" % get_ssh_common_args(conn)
+    conn.local(
+        "%s -P %s %s@%s:%s %s" % (cmd, conn.port, conn.user, conn.host, file, local)
+    )
+
+
+def manual_uboot_commit(conn):
+    conn.run("fw_setenv upgrade_available 0")
+    conn.run("fw_setenv bootcount 0")
+
+
+def common_board_setup(conn, files=None, remote_path="/tmp", image_file=None):
+    """
+    Deploy and activate an image to a board that uses mender-qa tools.
+
+    :param image_file: IMAGE_FILE as passed to deploy-test-image, can be None
+    :param remote_path: where files will be stored in the remote system
+    :param files: list of files to deploy
+    """
+    for f in files:
+        put_no_sftp(
+            os.path.basename(f),
+            conn,
+            remote=os.path.join(remote_path, os.path.basename(f)),
+        )
+
+    env_overrides = {}
+    if image_file:
+        env_overrides["IMAGE_FILE"] = image_file
+
+    conn.run(
+        "{} mender-qa deploy-test-image".format(
+            " ".join(["{}={}".format(k, v) for k, v in env_overrides.items()])
+        )
+    )
+
+    conn.sudo("mender-qa activate-test-image")
+
+
+def common_board_cleanup(conn):
+    conn.sudo("mender-qa activate-test-image off")
+    conn.sudo("reboot", warn=True)
+
+    run_after_connect("true", conn)
+
+
+def common_boot_from_internal(conn):
+    conn.sudo("mender-qa activate-test-image on")
+    conn.sudo("reboot", warn=True)
+
+    run_after_connect("true", conn)
+
+
+def latest_build_artifact(request, builddir, extension, sdimg_location=None):
```

Comment:
Edit: well done :) Leaving my comment below for historical reference (and to make sense of further comments)

--

I see that you have removed the global `configuration`.

```
@@ -296,13 +296,12 @@ def common_boot_from_internal(conn):
     run_after_connect("true", conn)
 
 
-def latest_build_artifact(builddir, extension, sdimg_location=None):
-    global configuration
+def latest_build_artifact(request, builddir, extension, sdimg_location=None):
 
     # Force the builddir to be an absolute path
     builddir = os.path.abspath(builddir)
 
-    if configuration.get("conversion"):
+    if request.config.getoption("--test-conversion"):
         output = subprocess.check_output(
             [
                 "sh",
```

Are you sure this works properly? This was introduced by Marcin P. at the time of the big refactor for Python 3, and I remember that he struggled a lot to get it working.

Of course I think your solution is cleaner, but it just smells like we are overlooking something...

Reply:
It works :)

Reply:
I regret doing it as part of this migration though, as this is what caused the most breakage! Always do one thing at a time, it's almost always faster!
RISC vs CISC basically, no? :)

Reply:
It's a personal approach, I think. You are young and very brave; I am quite conservative when it comes to moving pieces in test framework... stein på stein.